sdcm/utils/data_validator.py

```diff
@@ -469,6 +499,171 @@ def list_of_view_names_for_update_test(self):
+                severity=Severity.ERROR,
+                error="Analyzing of updated rows failed. Failed to get base table name from profile").publish()
+            return
+
+        result = {}
+        select_pk_columns = ', '.join(self.base_table_partition_keys)
+        for i, diff_row in enumerate(difference_set):
+            pks_for_missed_row = {}
+            missed_row_as_str = []
+            for pk_index, pk_name in enumerate(self.base_table_partition_keys):
+                pks_for_missed_row[pk_name] = diff_row[pk_index]
+                missed_row_as_str.append(f"{pk_name} = {diff_row[pk_index]}")
+
+            result[i] = {'row': ", ".join(missed_row_as_str)}
+            # data_for_validation.views[0] - view name with rows before update
+            # data_for_validation.views[1] - view name with rows after update
+            # data_for_validation.views[2] - view name with all expected partition keys
+            columns_for_validation = ', '.join(self.get_entity_columns(entity_name=data_for_validation.views[0],
+                                                                       session=session))
+            query = f"select %s f" \
+                    f"rom %s where {' and '.join(missed_row_as_str)}"
+
+            row_data = {}
+            for table in data_for_validation.views[:-1]:
+                if 'after_update' in table:
+                    key = 'after'
+                    select_columns = columns_for_validation
+                elif 'actual_data' in table:
+                    key = 'actual'
+                    select_columns = columns_for_validation
+                elif 'expect' in table:
+                    key = 'expect'
+                    select_columns = select_pk_columns
+                else:
+                    key = 'before'
+                    select_columns = columns_for_validation
+
+                try:
+                    row_data.update({key: list(session.execute(query % (select_columns, table)))})
+                except Exception as error:  # pylint: disable=broad-except
+                    LOGGER.error("Query %s failed. Error: %s", query % table, error)
+
+            row_data.update({'source_all_columns': list(session.execute(query % (columns_for_validation,
+                                                                                 self.base_table_name)))})
+            result[i].update({'rows': row_data})
+
+            if row_data['before'] or row_data['after']:
+                DataValidatorEvent.UpdatedRowsValidator(
+                    severity=Severity.WARNING,
+                    message=f"Row {diff_row} is found in the "
+                            f"{data_for_validation.views[0] if row_data['before'] else data_for_validation.views[1]} "
+                            f"view despite was not fetched. Test problem, failure in the fetcher\n"
+                            f"Row data in the source table: {row_data['source_all_columns']}\n"
```

Comment:
I don't think it's helpful to put all the data into an event, it too much information.

printing it in logs, maybe, but event are not designed to be that big (it won't be readable in email/argus)

Reply:
i.e. you can put the file names that are going to be saved. but not the whole data

Reply:
Changed it to debug message

Reply:
> i.e. you can put the file names that are going to be saved. but not the whole data

yes, I print the file name where analyzed data is saved